File: src/main/kotlin/com/yosep/server/infrastructure/component/sender/RedisMessageSender.java
package com.yosep.server.infrastructure.component.sender;

import org.redisson.api.RedissonReactiveClient;
import org.springframework.stereotype.Component;

import lombok.RequiredArgsConstructor;
import reactor.core.publisher.Mono;

@Component
@RequiredArgsConstructor
public class RedisMessageSender {

	private final RedissonReactiveClient redissonReactiveClient;

	public Mono<Long> send(String topic, String message) {
		return redissonReactiveClient.getTopic(topic).publish(message);
	}
}

File: src/main/kotlin/com/yosep/server/infrastructure/redis/component/sender/RedisMessageSender.kt
package com.yosep.server.infrastructure.redis.component.sender

import org.redisson.api.RedissonReactiveClient
import org.springframework.stereotype.Component
import kotlinx.coroutines.reactive.awaitSingle

@Component
class RedisMessageSender(
    private val redissonReactiveClient: RedissonReactiveClient
) {
    suspend fun send(topic: String, message: String): Long {
        return redissonReactiveClient.getTopic(topic)
            .publish(message)
            .awaitSingle()
    }
}